### PR TITLE
Feature -  5042 - Themed Accordions

### LINKED ISSUE #5042: Update theming of the Accordion component
## ✨ Feature

It's time to update the theming of our accordion component.

## 🕵️ Details

Currently the Accordion component is mostly just used with experiences and a lot of theming is built-in for that use case.  It also uses the old colour scheme.  We use the accordion on the Browse Opportunities page but we've continued to use the old theming for now.  We've received new designs for the Accordion.  They use an updated colour scheme and also have two different styles to use.

## ✅ Acceptance Criteria

- [ ] Refactor the existing Accordion component with a prop to change the styles
- [ ] Implement the two new styles demonstrated in the provided design: "card" and "simple"
     - [ ] Remove the existing "card with old colours" style
- [ ] Ensure spacing and layout conforms to provided design
- [ ] Storybook stories: 
    - [ ] "card" style
    - [ ] "simple" style
    - [ ] nesting, subtitle, and icon features are demonstrated
- [ ] Experience accordions use card style accordions
- [ ] Ongoing recruitment accordions use card style accordions
- [ ] Review from designer

Out of scope: Dark mode happens in #5532 

## 🎨 Design File

Reference design: https://github.com/GCTC-NTGC/gc-digital-talent/discussions/5522

### PR DESCRIPTION
**Pull request by esizer**
🤖 Resolves #5042

## 👋 Introduction

This updates the accordions to allow choosing a mode (simple, card) to use.

### ❗ Nested Accordion Issues

~~There are some issues with nesting the accordions right now. The issue is, the nested accordions are inheriting the styles from the root accordion so when we nest the card styles, we cannot use simple accordions inside. This may be my lack of experience with hydrogen.~~

~~I'm not sure how often we nest accordion and the priority of getting this working, do we have the time for me to get some assistance form @substrae on this? Or, do we want to break out the nested accordion styles into a new issue and sort that out at a later time once they have been completed?~~

## 🧪 Testing

Assist reviewers with steps they can take to test that the PR does what it says it does.

> **Warning**
> For the card style, there is a known issue with hydrogen `border-radius` so they are not rounded. This will be fixed with the upgrade to hydrogen in progress.

1. Run storybook `npm run storybook`
2. Navigate to the Accordion stories
3. Confirm they match the provided designs https://github.com/GCTC-NTGC/gc-digital-talent/discussions/5522

### Card

Screenshot 2023-01-31 at 4 14 09 PM

### Simple

Screenshot 2023-01-31 at 4 14 22 PM

### Nested

Screenshot 2023-02-01 120015

### CODE REVIEW

**Comment by substrae** [comment]
Hmm, this shouldn't be a Hydrogen issue unless you're using the `:selectors[]` or `:children[]` and specifying selector values that are vague enough that they target the nested accordions too. I can't look at the code in the next few days, but if you can respond here with more specifics I might be able to help.

**Comment by esizer** [comment]
> Hmm, this shouldn't be a Hydrogen issue unless you're using the `:selectors[]` or `:children[]` and specifying selector values that are vague enough that they target the nested accordions too. I can't look at the code in the next few days, but if you can respond here with more specifics I might be able to help.

That is exactly what we are doing here. The theme is set at the root and we use `:children[]` to select. I may need to update to use more direct selectors like `:selectors[>.class]` to only target direct children rather than _any_ child 🤔

**Comment by substrae** [comment]
Yep, that would be my recommendation too. Same idea if we were applying regular ol' CSS.

**Comment by esizer** [comment]
> Yep, that would be my recommendation too. Same idea if we were applying regular ol' CSS.

Thanks! That worked 🎉

**Review by petertgiles — CHANGES_REQUESTED** [review]
Wow, another work of art!  :exploding_head: 

I still think you should get a designer to review before merging.  Maybe Jerbo would be willing in Josh's absence?

One small thing to check below.

**Comment by petertgiles** [comment]
@Jerryescandon would you be willing to provide a designer review in Josh's absence?
https://www.chromatic.com/component?appId=61e099a1bb1465003a73d3ce&csfId=components-accordion&buildNumber=4260&k=63da9ec1a06a8f8f9953fc63-1200-interactive-true&h=50&b=0

**Comment by Jerryescandon** [comment]
Oh wao, big responsibility, Imma be as nitpicky as possible just to avoid letting Josh down. 

Number 1 nitpickiness is target area. I feel like its important for the full box to be clickable and theres a couple places where it aint. 🤔 
chrome_jL3KtqKVLU

**Comment by Jerryescandon** [comment]
Second nitpickyness is the space between items in the simple version. It should be doing 2 things, lowering the padding by half, and removing the space between the items. I think we are reducing the padding but it feels like the space between items is still there cause the list doesn't feel as small as it could be.

image


image

**Comment by Jerryescandon** [comment]
That's all I can think of 👍

File: frontend/common/src/components/Accordion/Accordion.tsx [hunk 1]
```diff
@@ -96,14 +99,25 @@ const Trigger = React.forwardRef<
...
           <StyledTrigger ref={forwardedRef} {...rest}>
-            <div data-h2-margin="base(0, x.25, 0, 0)" style={{ flexShrink: 0 }}>
+            <div
+              className="Accordion__Chevron"
+              data-h2-display="base(flex)"
+              data-h2-align-items="base(center)"
+              data-h2-flex-shrink="base(0)"
+              data-h2-radius="base(circle)"
+            >
               <ChevronRightIcon
```
[the remark this answers is not in the document]
Comment (petertgiles) [reply frontend/common/src/components/Accordion/Accordion.tsx]: Looks amazing!  :medal_sports:

**Review by petertgiles — APPROVED** [review]

**Pull request merged**